oeps/oep-0044-bp-changelogs.rst

```diff
@@ -0,0 +1,285 @@
+
+If you find a backward incompatible change in a MINOR or PATCH version after publishing the package, please update the changelog and communicate the inconsistency. Additionally, you have the following options:
+
+* A new MINOR or PATCH version can be immediately released to revert the change, and the change can be restored with a new MAJOR version. Or,
+* If you decide that reverting will cause more harm for some reason, and you do not intend to revert, please let this be a part of your communication.
+
+The reason it is important that only MAJOR versions include backward incompatible change is because when people upgrade dependencies, some will only read changelogs for MAJOR versions because of the potential backward incompatibilities, and may not read notes for MINOR or PATCH upgrades.
+
+Frontend Libraries
+~~~~~~~~~~~~~~~~~~
+
+Like all Open edX repositories, frontend libraries should also include a CHANGELOG.rst. However, as you'll see below, this file will simply point to the GitHub Releases. Here is an example CHANGELOG.rst::
+
+  Changelog
+  =========
+
+  All notable changes to this project will be documented as `GitHub Releases`_.
+
+  This project adheres to `Semantic Versioning`_.
+
+  .. _GitHub Releases: https://github.com/edx/frontend-build/releases
+  .. _Semantic Versioning: https://semver.org/spec/v2.0.0.html
+
+
+All new frontend libraries have been using an npm package called `semantic-release`_.
+
+The `semantic-release`_ package automates the following:
+
+* Incrementing the version appropriately for each PR merge, based on a well formatted commit comment. See the `semantic-release`_ README for details. A pre-commit hook is typically run to ensure the commit message is appropriately formatted for `semantic-release`_.
+
+  * One important formatting detail is to include "BREAKING CHANGE" at the beginning of a line in the commit description to get a new major version.
+
+* Publishing the changelog message as a GitHub Release in the repository, using the commit message. Here is an example `GitHub Release changelog for edx/frontend-build`_.
+
+As noted above, please update the changelogs as necessary to add clarifications or customize the messaging for the audience of the changelog, especially because these messages originally come from commit messages, which may have been written for a different audience.
+
+.. _semantic-release: https://github.com/semantic-release/semantic-release
+.. _GitHub Release changelog for edx/frontend-build: https://github.com/edx/frontend-build/releases
+
+Python Libraries
+~~~~~~~~~~~~~~~~
+
+Like all Open edX repositories, Python libraries should also include a CHANGELOG.rst. This file may also simply point to the GitHub Releases. Here is an example CHANGELOG.rst::
```

Comment:
Without extra tooling GitHub Releases are *not* a good changelog solution.  They provide no help for writing the changelog.

Reply:
Can you clarify what you mean?  Here are the Github releases for edx-rest-api-client, as an example:  https://github.com/edx/edx-rest-api-client/releases.  When you publish the release notes, the library is published to PyPI.  Is that "extra tooling"?  If not, can you clarify what "extra tooling" means to you, and why this current process is "_not_ a good changelog solution."  Note: for me as the audience of these changelog entries, they also have always been sufficient for my needs during upgrades.

Reply:
In my testing, the release shows only the latest commit by default.  You get a blank text field to write into if you want, but do we think people will write new text there? I thought we were trying to reduce the duplicated work.  If people are willing to write separately about the changes, why not do it in a CHANGELOG file?

If there's more than one change in a release, GitHub releases doesn't seem to help to summarize them all.  Are we assuming that every commit will become a release?

Reply:
By "extra tooling" I meant something like the "semantic-releases" tool the frontend libraries are using, which populates the release description with commit messages.